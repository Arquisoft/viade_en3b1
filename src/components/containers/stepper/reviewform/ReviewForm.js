import React, { Component } from 'react';
import { Typography, Grid, Button, TextField } from '@material-ui/core';

export class ReviewForm extends Component {

    next = (e) => {
        e.preventDefault();
        this.props.createRoute();
        this.props.handleNext();
    }

    back = (e) => {
        e.preventDefault();
        this.props.handleBack();
    }

    getPointsText(points){
        var pointsText = "";

        points.forEach((p) => pointsText += "Lat: "+p.lat+"\tLong: "+p.lng+"\n");

        return pointsText;
    }

    render() {

        const { values } = this.props;

        var month = values.date.getUTCMonth() + 1; //months from 1-12
        var day = values.date.getUTCDate();
        var year = values.date.getUTCFullYear();

        const date = month + "/" + day + "/" + year;        

        return (
            <div>
                <React.Fragment>
                    <Typography variant="h6" gutterBottom>
                        Please, review all the data
                    </Typography>

                    <Grid container spacing={3}>
                        <Grid item xs={12} sm={6} style={{ marginTop: 16 }}>
                            <TextField
                                disabled
                                id="name"
                                name="name"
                                label="Name"
                                fullWidth
                                defaultValue={values.name}
                                variant="filled"
                            />
                        </Grid>

                        <Grid item xs={12} sm={6} style={{ marginTop: 16 }}>
                            <TextField
                                disabled
                                id="date"
                                name="date"
                                label="Date"
                                fullWidth
                                defaultValue={date}
                                variant="filled"
                            />
                        </Grid>

                        <Grid item xs={12}>
                            <TextField
                                disabled
                                id="description"
                                name="description"
                                label="Description"
                                multiline
                                fullWidth
                                defaultValue={values.description}
                                rows="4"
                                variant="filled"
                            />
                        </Grid>

                        <Grid item xs={12}>
                            <TextField
                                disabled
                                id="mapPoint"
                                name="mapPoint"
                                label="Track Points"
                                multiline
                                fullWidth
                                defaultValue={this.getPointsText(values.points)}
                                rows="4"
                                variant="filled"
                            />
                        </Grid>
                    </Grid>

                    <Grid container spacing={3} style={{ marginTop: 12 }}>
                        <Button
                            variant="contained"
                            color="default"
                            style={{
                                marginTop: 12,
                                marginLeft: '1vh'
                            }}
                            onClick={this.back}
                        >
                            Back
                        </Button>

                        <Button
                            variant="contained"
                            color="primary"
                            style={{
                                marginTop: 12,
                                marginLeft: 'auto'
                            }}
                            onClick={this.next}
                        >
                            Next
                        </Button>
                    </Grid>
                </React.Fragment>
            </div>
        );
    }
}

export default ReviewForm;
