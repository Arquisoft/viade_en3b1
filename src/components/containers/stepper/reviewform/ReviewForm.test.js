import ReviewForm from './ReviewForm';

const state = {
  activeStep: 0,
  name: '',
  description: '',
  date: new Date(),
  photos: [],
  videos: [],
  points: []
}

const { activeStep, name, description, date, photos, videos, points } = state;
const values = { activeStep, name, description, date, photos, videos, points };

const handleNext = jest.fn();
const handleBack = jest.fn();
const handleChange = jest.fn();
const handleDateChange = jest.fn();
const handleMediaChange = jest.fn();
const handleMapPoints = jest.fn();
const handleDownload = jest.fn();
const createRoute = jest.fn();

test('Define ReviewForm', () => {
  expect(ReviewForm).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<ReviewForm
    handleNext={handleNext}
    handleBack={handleBack}
    values={values}
    createRoute={createRoute}
/>);
  expect(wrapper).toMatchSnapshot();
});