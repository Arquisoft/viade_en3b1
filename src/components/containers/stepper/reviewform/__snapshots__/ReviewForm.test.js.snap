// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Please, review all the data
  </h6>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style="margin-top:16px"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
          for="name"
          id="name-label"
        >
          Name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
            disabled=""
            id="name"
            name="name"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style="margin-top:16px"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-disabled Mui-disabled MuiFormLabel-filled"
          data-shrink="true"
          for="date"
          id="date-label"
        >
          Date
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
            disabled=""
            id="date"
            name="date"
            type="text"
            value="4/29/2020"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
          for="description"
          id="description-label"
        >
          Description
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            disabled=""
            id="description"
            name="description"
            rows="4"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
          for="mapPoint"
          id="mapPoint-label"
        >
          Track Points
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            disabled=""
            id="mapPoint"
            name="mapPoint"
            rows="4"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    style="margin-top:12px"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      style="margin-top:12px;margin-left:1vh"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      id="btn-next"
      style="margin-top:12px;margin-left:auto"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
    </button>
  </div>
</div>
`;
